backend/isoko-backend/src/handlers/business/post-list-business.js

```diff
@@ -9,10 +14,79 @@ exports.postListBusinessHandler = async (event) => {
    }
 
    console.info('received:', event);
+   const requestBody = event.body && JSON.parse(event.body);
+   
+   const name = _.get(requestBody, 'name');
+   // add defaultVal param bc city is optional 
+   const city = _.get(requestBody, 'city', "");
+   const type = _.get(requestBody, 'type');
+   const tags = _.get(requestBody, 'tags');
+   const keywords = _.get(requestBody, 'keywords');
+   const shortDesc = _.get(requestBody, 'shortDesc');
+   const businessId = _.get(requestBody, 'businessId');
+   const hours = _.get(requestBody, 'hours');
+   const links = _.get(requestBody, 'links');
+   const address = _.get(requestBody, 'address');
+   const aboutOwner = _.get(requestBody, 'aboutOwner', null);
+   const owner = _.get(aboutOwner, 'owner', "");
+   const ownerName = _.get(aboutOwner, 'ownerName', ""); 
+   const ownerPhone = _.get(aboutOwner, 'ownerPhone', "");
+   const ownerDesc = _.get(aboutOwner, 'ownerDesc', "");
+   const photo = _.get(aboutOwner, 'photo', "");
+   const lister = _.get(requestBody, 'lister');
+   
+
+   const queryParams = {
+      TableName: BUSINESS_TABLE,
+      Item: {
+         "name": name, 
+         "city": city,
+         "type": type, 
+         "tags": tags, 
+         "keywords": keywords,
+         "shortDesc": shortDesc, 
+         "businessId": businessId, 
+         "hours": hours, 
+         "links": links, 
+         "address": address,
+         "aboutOwner": {
+            "owner": owner,
+            "ownerName": ownerName, 
+            "ownerPhone": ownerPhone, 
+            "ownerDesc": ownerDesc,
+            "photo": photo, 
+         }, 
+         "lister": lister
+      }
+   }
+
+   // remove optional params if not present
+   if (city == "") {
+      delete queryParams.Item.city; 
+   }
+   
+   // if request is coming from non-owner list a business page, following params will not be included 
+   if (owner == "") {
+         delete queryParams.Item.aboutOwner.owner;
+         delete queryParams.Item.aboutOwner.ownerDesc; 
+         delete queryParams.Item.aboutOwner.photo;  
+         delete queryParams.Item.shortDesc; 
+         delete queryParams.Item.hours; 
+         delete queryParams.Item.links; 
+   }
```

Comment:
Kind of like what Rohith did in his task, I don't think we actually need to delete these parameters as long as they're undefined when we put them into Dynamodb

Reply:
got it. yeah i'm pretty sure as long as you specify a default value, like "" for a string, it just stores that. i wasn't sure if we wanted empty vals in the db like that or not and i wrote this before looking at rohith's stuff lol